test/generic/ModuleHomomorphism-test.jl

```diff
@@ -1,197 +1,197 @@
 
-      Q, g = quo(M, M1)
-      k1, h1 = kernel(g)
+    Q, g = quo(M, M1)
+    k1, h1 = kernel(g)
 
-      k = compose(h1, g)
-      I, f = image(k)
-      T, t = sub(Q, elem_type(Q)[])
+    k = compose(h1, g)
+    I, f = image(k)
+    T, t = sub(Q, elem_type(Q)[])
 
-      @test I == T
+    @test I == T
 
-      Q, g = quo(M, M)
-      @test all(x -> iszero(g(x)), gens(Q))
-   end
+    Q, g = quo(M, M)
+    @test all(x -> iszero(g(x)), gens(Q))
+  end
 
-   S = AbstractAlgebra.JuliaQQ
-   for iter = 1:100
-      # test image of composition of canonical injection and projection
-      N = rand_module(S, -10:10)::AbstractAlgebra.FPModule{elem_type(S)}
+  S = AbstractAlgebra.JuliaQQ
+  for iter = 1:100
+    # test image of composition of canonical injection and projection
+    N = rand_module(S, -10:10)::AbstractAlgebra.FPModule{elem_type(S)}
 
-      ngens1 = rand(1:5)
-      gens1 = [rand(N, -10:10) for j in 1:ngens1]
-      M1, f1 = sub(N, gens1)
+    ngens1 = rand(1:5)
+    gens1 = [rand(N, -10:10) for j in 1:ngens1]
+    M1, f1 = sub(N, gens1)
 
-      Q, g = quo(N, M1)
-      k1, h1 = kernel(g)
+    Q, g = quo(N, M1)
+    k1, h1 = kernel(g)
 
-      k = compose(h1, g)
-      I, f = image(k)
-      T, t = sub(Q, elem_type(Q)[])
+    k = compose(h1, g)
+    I, f = image(k)
+    T, t = sub(Q, elem_type(Q)[])
 
-      @test I == T
+    @test I == T
 
-      Q, g = quo(N, N)
-      @test all(x -> iszero(g(x)), gens(Q))
-   end
+    Q, g = quo(N, N)
+    @test all(x -> iszero(g(x)), gens(Q))
+  end
 end
 
 @testset "Generic.ModuleHomomorphism.preimage" begin
-   for R in [ZZ, QQ]
-      for iter = 1:5
-         N = rand_module(R, -10:10)
-
-         ngens1 = rand(1:5)
-         gens1 = [rand(N, -10:10) for j in 1:ngens1]
-         M, f = sub(N, gens1)
-
-         m = rand(M, -10:10)
-         n = f(m)
-         x = preimage(f, n)
-         @test x == m
-
-         for i in 0:5
-            m = elem_type(M)[ rand(M, -10:10) for j in 1:i ]
-            n = elem_type(N)[ f(mm) for mm in m ]
-            x = preimage(f, n)
-            @test x == m
-         end
+  for R in [ZZ, QQ]
+    for iter = 1:5
+      N = rand_module(R, -10:10)
+
+      ngens1 = rand(1:5)
+      gens1 = [rand(N, -10:10) for j in 1:ngens1]
+      M, f = sub(N, gens1)
+
+      m = rand(M, -10:10)
+      n = f(m)
+      x = preimage(f, n)
+      @test x == m
+
+      for i in 0:5
+        m = elem_type(M)[ rand(M, -10:10) for j in 1:i ]
+        n = elem_type(N)[ f(mm) for mm in m ]
+        x = preimage(f, n)
+        @test x == m
       end
-   end
+    end
+  end
 end
 
 @testset "Generic.ModuleHomomorphism.printing" begin
   M = free_module(ZZ, 2)
   f = ModuleHomomorphism(M, M, matrix(ZZ, 2, 2, [1, 2, 3, 4]))
   str = """
-        Module homomorphism
-          from free module of rank 2 over integers
-          to free module of rank 2 over integers"""
+  Module homomorphism
+  from free module of rank 2 over integers
+  to free module of rank 2 over integers"""
```

Comment:
This change is wrong!

Reply:
Yes, the tests are also failing. I look at it later.

Reply:
Should be fine now.